src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -149,6 +201,8 @@ public void removeTaskId(String taskId) {
         if (entry.getValue() != null && entry.getValue().equals(taskId)) {
           journalNodes.put(entry.getKey(), null);
           setJournalNodes(journalNodes);
+          Date date = DateUtils.addSeconds(new Date(), conf.getDeadNodeTimeout());
+          deadJournalNodeTimeStamp = new Timestamp(date.getTime());
```

Comment:
"if you keep getting a new DN failure every minute, and only half of them recover, you won't ever remove any of the first failures." Or more concretely:
T=0: DN1 fails (and will never recover); HDFS starts the timeout to delay scheduling DNs on new hosts.
T=1min: DN2 fails and resets the timeout (but eventually/quickly recovers); HDFS still waiting for the timeout before clearing deadDNs and launching DNs on new hosts.
T=2...: A new DN fails (and recovers) every minute, continually resetting the timeout; HDFS will never clear deadDNs and will never be able to launch DNs on new hosts.
So, even though only one DN is actually dead, and has been dead for minutes/hours, the timeout never completes, and that old dead DN is never removed.